crates/bevy_render/src/texture/compressed_image_saver.rs

```diff
@@ -0,0 +1,51 @@
+use crate::texture::{Image, ImageFormat, ImageFormatSetting, ImageLoader, ImageLoaderSettings};
+use bevy_asset::saver::{AssetSaver, SavedAsset};
+use futures_lite::{AsyncWriteExt, FutureExt};
+
+pub struct CompressedImageSaver;
+
+impl AssetSaver for CompressedImageSaver {
+    type Asset = Image;
+
+    type Settings = ();
+    type OutputLoader = ImageLoader;
+
+    fn save<'a>(
+        &'a self,
+        writer: &'a mut bevy_asset::io::Writer,
+        image: SavedAsset<'a, Self::Asset>,
+        _settings: &'a Self::Settings,
+    ) -> bevy_utils::BoxedFuture<'a, std::result::Result<ImageLoaderSettings, anyhow::Error>> {
+        // PERF: this should live inside the future, but CompressorParams and Compressor are not Send / can't be owned by the BoxedFuture (which _is_ Send)
```

Comment:
I guess that should be discussed/resolved in a future PR ; but I recently saw a (unsafe) wrapper to force `Send`, that pattern may useful here: https://github.com/rust-windowing/winit/blob/309e6aa85ae60104308d039763d28f42d6fe1595/src/window.rs#L1483C4-L1483C4

Reply:
We'd need to do a safety investigation first, but worth looking into. I agree its probably a "future pr" thing.